src/Log/Formatter/SyslogFormatter.php

```diff
@@ -0,0 +1,47 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) :  Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakefoundation.org CakePHP(tm) Project
+ * @since         4.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Log\Formatter;
+
+class SyslogFormatter extends AbstractFormatter
+{
+    /**
+     * Default config for this class
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'format' => '%s: %s',
+    ];
+
+    /**
+     * @param array $config Formatter config
+     */
+    public function __construct(array $config = [])
+    {
+        $this->setConfig($config);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function format($level, string $message, array $context = []): array
+    {
+        return array_map(function ($line) use ($level) {
+            return sprintf($this->getConfig('format'), $level, $line);
+        }, explode("\n", $message));
```

Comment:
Should I make `SyslogEngine` split the message into lines after resolving the placeholders, and call `format()` for each one instead? This is the main reason why `format()` returns `array`. I don't know that other formatters would need to split up the message.

Reply:
I think splitting the message up into multiple lines is something the log engine can do. Having that in the formatter makes formatters less reusable.